### PR TITLE
Upgrade documentation includes notes about storage version migration

### PR DESCRIPTION
**Pull request by dprotaso**
Documentation for https://github.com/knative/serving/issues/6726

## Proposed Changes 

- Include documentation that users may be required to migrate resources to the latest storage

### CODE REVIEW

File: docs/install/upgrade-installation.md [hunk 1]
```diff
@@ -59,6 +59,23 @@ If you have a plug-in installed, make sure to upgrade it at the same time as
 you upgrade your Knative components. For example, if you have the
 monitoring plug-in installed, upgrade it alongside Knative Serving and Eventing.
 
+
+### Upgrade existing resources to the latest stored version
+
+Our custom resources are stored within Kubernetes at a particular version.
```
Comment (mattmoor) [inline docs/install/upgrade-installation.md:65]: Remove trailing whitespace:
```suggestion
Our custom resources are stored within Kubernetes at a particular version.
```

File: docs/install/upgrade-installation.md [hunk 2]
```diff
@@ -59,6 +59,23 @@ If you have a plug-in installed, make sure to upgrade it at the same time as
 you upgrade your Knative components. For example, if you have the
 monitoring plug-in installed, upgrade it alongside Knative Serving and Eventing.
 
+
+### Upgrade existing resources to the latest stored version
+
+Our custom resources are stored within Kubernetes at a particular version. 
+As we introduce newer and remove older versions you'll need to migrate our resources
```
Comment (mattmoor) [inline docs/install/upgrade-installation.md:66]: Remove trailing whitespace:
```suggestion
As we introduce newer and remove older versions you'll need to migrate our resources
```

File: docs/install/upgrade-installation.md [hunk 3]
```diff
@@ -59,6 +59,23 @@ If you have a plug-in installed, make sure to upgrade it at the same time as
 you upgrade your Knative components. For example, if you have the
 monitoring plug-in installed, upgrade it alongside Knative Serving and Eventing.
 
+
+### Upgrade existing resources to the latest stored version
+
+Our custom resources are stored within Kubernetes at a particular version. 
+As we introduce newer and remove older versions you'll need to migrate our resources 
+to the designated stored version. This ensures removing older verions
```
Comment (mattmoor) [inline docs/install/upgrade-installation.md:67]: Remove trailing whitespace:
```suggestion
to the designated stored version. This ensures removing older verions
```

File: docs/install/upgrade-installation.md [hunk 4]
```diff
@@ -59,6 +59,23 @@ If you have a plug-in installed, make sure to upgrade it at the same time as
...
+As we introduce newer and remove older versions you'll need to migrate our resources
+to the designated stored version. This ensures removing older versions
+will succeed when upgrading.
+
+For the various subprojects - we have a K8s job to help operators perform this migration.
+The release notes for each release will explicitly whether a migration is required.
+
+ie.
+```bash
+kubectl apply --filename https://github.com/knative/serving/releases/download/v0.14.0/serving-storage-version-migration.yaml
```
Comment (mattmoor) [inline docs/install/upgrade-installation.md:75]: Use the `{{ artifact }}` shortcode

File: docs/install/upgrade-installation.md [hunk 5]
```diff
@@ -72,7 +72,7 @@ The release notes for each release will explicitly whether a migration is requir
 
 ie.
 ```bash
-kubectl apply --filename https://github.com/knative/serving/releases/download/v0.14.0/serving-storage-version-migration.yaml
+kubectl apply --filename kubectl apply --filename {{< artifact repo="serving" file="serving-storage-version-migration.yaml" >}}
```
Comment (mattmoor) [inline docs/install/upgrade-installation.md:75]: this looks messed up :)

Comment (mattmoor) [reply docs/install/upgrade-installation.md:75]:
```suggestion
kubectl apply --filename {{< artifact repo="serving" file="serving-storage-version-migration.yaml" >}}
```

**Comment by dprotaso** [comment]
🤦‍♂️

**Pull request merged**